packages/core/src/modules/dids/methods/key/KeyDidRegistrar.ts

```diff
@@ -39,10 +41,22 @@ export class KeyDidRegistrar implements DidRegistrar {
       }
     }
 
+    if (privateKey && (typeof privateKey !== 'object' || privateKey.length !== 32)) {
```

Comment:
```suggestion
    if (privateKey && (!Buffer.isBuffer(privateKey) || privateKey.length !== 32) {
```

Reply:
I'm not sure if the privateKey always needs to be 32 bytes? It depends on the key type I guess, so maybe we need to handle this in the wallet?